framework/retryresource/resource.go

```diff
@@ -85,7 +90,7 @@ func (r *Resource) GetCurrentState(obj interface{}) (interface{}, error) {
 	}
 
 	n := func(err error, dur time.Duration) {
-		r.logger.Log("warning", fmt.Sprintf("retrying 'GetCurrentState' due to error (%s)", err.Error()))
+		r.logger.Log("warning", fmt.Sprintf("retrying 'GetCurrentState' of resource '%s' due to error (%s)", r.Origin().Name(), err.Error()))
```

Comment:
A nice side effect with the interface extensions is that the logged warnings are actually way more useful.